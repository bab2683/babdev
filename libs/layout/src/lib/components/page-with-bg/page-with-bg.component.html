<div class="page-with-bg">
  <div class="title-container" [ngClass]="{ 'title-container--mobile': isMobile }">
    <div
      class="title-container__mobile-bg"
      *ngIf="bgPath && isMobile"
      [ngStyle]="{
        'background-image': 'url(' + path + ')',
        'background-position-x': positionX,
        'background-position-y': positionY
      }"
    ></div>
    <h1 class="title-container__title">
      {{ title }}
    </h1>
  </div>
  <div class="content">
    <ng-content></ng-content>
  </div>
</div>
<div
  class="bg"
  *ngIf="bgPath && !isMobile"
  [ngStyle]="{
    'background-image': 'url(' + path + ')',
    'background-position-x': positionX,
    'background-position-y': positionY
  }"
></div>
