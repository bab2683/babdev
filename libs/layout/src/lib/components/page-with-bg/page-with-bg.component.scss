@import "~@styleguide_funcional";

:host {
  display: flex;
  min-height: 100%;
}

@mixin absolute-full {
  height: 100%;
  left: 0;
  position: absolute;
  top: 0;
  width: 100%;
}

$title-container-selector: ".title-container";

#{$title-container-selector} {
  position: relative;

  &--mobile {
    align-items: center;
    display: flex;
    justify-content: center;
    min-height: 30vh;

    #{$title-container-selector}__title {
      color: color("white");
      position: relative;
    }
  }

  &__title {
    padding: 0 var(--baddev-content-padding);
  }

  &__mobile-bg {
    background-size: cover;

    @include absolute-full;

    &::after {
      background-color: color("main");
      content: "";
      opacity: 0.7;

      @include absolute-full;
    }
  }
}

.page-with-bg {
  flex: 1;
}

.content {
  flex: 1;
  padding: var(--baddev-content-padding);
}

.bg {
  background-size: cover;
  min-width: px-to-rem(300);
}
