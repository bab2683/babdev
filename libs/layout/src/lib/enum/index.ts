export * from './background.enum';
export * from './clickable.enum';
export * from './tooltip.enum';
