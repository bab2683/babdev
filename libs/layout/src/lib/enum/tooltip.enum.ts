export enum TooltipStatus {
  CLOSED = 'closed',
  OPEN = 'open'
}
