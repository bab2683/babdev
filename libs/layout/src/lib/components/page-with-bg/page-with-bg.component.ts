import { Component, Input, OnInit } from '@angular/core';

import { BackGroundPositionXEnum, BackGroundPositionYEnum } from '../../enum';

@Component({
  selector: 'babdev-page-with-bg',
  templateUrl: './page-with-bg.component.html',
  styleUrls: ['./page-with-bg.component.scss']
})
export class PageWithBgComponent implements OnInit {
  @Input() public assetsPath: string = 'assets/';
  @Input() public isMobile: boolean = false;
  @Input() public bgPath: string;
  @Input() public bgPositionX: BackGroundPositionXEnum | number = BackGroundPositionXEnum.CENTER;
  @Input() public bgPositionY: BackGroundPositionYEnum | number = 0;
  @Input() public format: string = 'jpg';

  public path: string;
  public positionX: BackGroundPositionXEnum | string;
  public positionY: BackGroundPositionXEnum | string;

  public ngOnInit(): void {
    this.path = `/${this.assetsPath}${this.bgPath}.${this.format}`;
    this.positionX =
      typeof this.bgPositionX === 'number' ? `${this.bgPositionX}%` : this.bgPositionX;
    this.positionY =
      typeof this.bgPositionY === 'number' ? `${this.bgPositionY}%` : this.bgPositionY;
  }
}
