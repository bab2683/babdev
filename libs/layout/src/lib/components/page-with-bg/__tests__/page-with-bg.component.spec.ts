import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { BackGroundPositionXEnum, BackGroundPositionYEnum } from '../../../enum';
import { PageWithBgComponent } from '../page-with-bg.component';

describe('PageWithBgComponent', () => {
  let component: PageWithBgComponent;
  let fixture: ComponentFixture<PageWithBgComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [PageWithBgComponent]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(PageWithBgComponent);
    component = fixture.componentInstance;
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match snapshot', () => {
    component.bgPath = 'my_image_path';
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should match snapshot', () => {
    component.bgPath = 'my_image_path';
    component.bgPositionX = 10;
    component.bgPositionY = BackGroundPositionYEnum.TOP;
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should match snapshot', () => {
    component.isMobile = true;
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});
