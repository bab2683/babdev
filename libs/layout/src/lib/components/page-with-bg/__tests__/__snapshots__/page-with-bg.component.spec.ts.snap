// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageWithBgComponent should match snapshot 1`] = `
<babdev-page-with-bg
  assetsPath={[Function String]}
  bgPath={[Function String]}
  bgPositionX={[Function String]}
  bgPositionY="0"
  format={[Function String]}
  isMobile="false"
  path={[Function String]}
  positionX={[Function String]}
  positionY={[Function String]}
>
  <div
    class="page-with-bg"
  >
    <div
      class="title-container"
    >
      
      <h1
        class="title-container__title"
      >
          
      </h1>
    </div>
    <div
      class="content"
    />
  </div><div
    class="bg"
    style="background-image: url(/assets/my_image_path.jpg); background-position-x: center; background-position-y: 0%;"
  />
</babdev-page-with-bg>
`;

exports[`PageWithBgComponent should match snapshot 2`] = `
<babdev-page-with-bg
  assetsPath={[Function String]}
  bgPath={[Function String]}
  bgPositionX={[Function Number]}
  bgPositionY={[Function String]}
  format={[Function String]}
  isMobile="false"
  path={[Function String]}
  positionX={[Function String]}
  positionY={[Function String]}
>
  <div
    class="page-with-bg"
  >
    <div
      class="title-container"
    >
      
      <h1
        class="title-container__title"
      >
          
      </h1>
    </div>
    <div
      class="content"
    />
  </div><div
    class="bg"
    style="background-image: url(/assets/my_image_path.jpg); background-position-x: 10%; background-position-y: top;"
  />
</babdev-page-with-bg>
`;

exports[`PageWithBgComponent should match snapshot 3`] = `
<babdev-page-with-bg
  assetsPath={[Function String]}
  bgPositionX={[Function String]}
  bgPositionY="0"
  format={[Function String]}
  isMobile={[Function Boolean]}
  path={[Function String]}
  positionX={[Function String]}
  positionY={[Function String]}
>
  <div
    class="page-with-bg"
  >
    <div
      class="title-container title-container--mobile"
    >
      
      <h1
        class="title-container__title"
      >
          
      </h1>
    </div>
    <div
      class="content"
    />
  </div>
</babdev-page-with-bg>
`;
